File: Server/route/challenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetChallengeRoute(r *gin.Engine) {
	challenge := r.Group("/challenge")
	challenge.Use(middleware.JWTAuthMiddleware())
	{
		challenge.POST("/GetChallengeList", api.GetChallengeList)
		challenge.POST("/GetChallenge", api.GetChallenge)
	}
	adminChallenge := r.Group("/challenge")
	adminChallenge.Use(middleware.AdminAuthMiddleware())
	{
		adminChallenge.POST("/CreateChallenge", api.CreateChallenge)
		adminChallenge.POST("/UpdateChallenge", api.UpdateChallenge)
		adminChallenge.POST("/DeleteChallenge", api.DeleteChallenge)
	}
}

File: Web/src/api/challenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetChallengeList(data) {
    return apiClient.post("/challenge/GetChallengeList", data).then(handleResponse).catch(handleError)
}

function GetChallenge(data) {
    return apiClient.post("/challenge/GetChallenge", data).then(handleResponse).catch(handleError)
}

function CreateChallenge(data) {
    return apiClient.post("/challenge/CreateChallenge", data).then(handleResponse).catch(handleError)
}

function UpdateChallenge(data) {
    return apiClient.post("/challenge/UpdateChallenge", data).then(handleResponse).catch(handleError)

}

function DeleteChallenge(data) {
    return apiClient.post("/challenge/DeleteChallenge", data).then(handleResponse).catch(handleError)

}

const challengeApi = {
  GetChallengeList,
  GetChallenge,
  CreateChallenge,
  UpdateChallenge,
  DeleteChallenge
}

export default challengeApi
